### DISCUSSION TITLE
AWS CLI help with ec2

### DISCUSSION THREAD

**Discussion by johnxlau** [aws/aws-cli#6913] [Q&A] [answered]
when I run the command aws s3 ls it produce the list of s3 buckets, but
when I run the command: aws ec2 describe-instances 
it produce below error:

Failed to connect to proxy URL: http://proxy.abc.com/

What is wrong with setting, aws cli is installed on Linux.

**Comment by tim-finnigan** [comment, marked as answer]
Hi @johnxlau thanks for reaching out. Have you configured your CLI (setting your region/credentials)? Here is [documentation](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html) that can help with that.

And if you are using proxies you can refer to [this documentation](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-proxy.html).

**Comment by johnxlau** [comment]
Hi Tim,

I think I did....
I must be miss something regarding ec2 in aws...

export HTTP_PROXY=http://proxy.mycompany.com
export HTTPS_PROXY=http://proxy.mycompany.com
export AWS_ACCESS_KEY_ID=xxxx
export AWS_SECRET_ACCESS_yyy
export AWS_SESSION_TOKEN=zzzz
export AWS_DEFAULT_REGION=us-east-1
export NO_PROXY=169.254.169.254

$ aws s3 ls
2021-02-18 02:49:09 ansible-demo-bucket
2022-03-07 09:39:59 cvw-claimaint
2022-02-02 05:25:25 cvw-clent

$ aws ec2 describe-instances

Failed to connect to proxy URL: "http://proxy.mycompany.com"

**Reply by tim-finnigan** [reply]
Is there an issue with your proxy URL or network?

**Reply by johnxlau** [reply]
Why is that I do not have problem with aws s3 commands, but only with aws ec2 commands?  It goes thru the same proxy url.
The linux machine that I am using is behind corp network.  It is frustrating.

**Reply by tim-finnigan** [reply]
Can you provide the full debug logs by adding `--debug` to your command? Please redact any sensitive information.

Also can you confirm that you don't have any firewall settings configured that might be causing the issue?
